clients/da-vinci-client/src/main/java/com/linkedin/davinci/client/AvroGenericDaVinciClient.java

```diff
@@ -719,7 +721,7 @@ public synchronized void start() {
     logger.info("Starting client, storeName=" + getStoreName());
     VeniceConfigLoader configLoader = buildVeniceConfig();
     Optional<ObjectCacheConfig> cacheConfig = Optional.ofNullable(daVinciConfig.getCacheConfig());
-    initBackend(clientConfig, configLoader, managedClients, icProvider, cacheConfig);
+    initBackend(clientConfig, configLoader, managedClients, icProvider, cacheConfig, recordTransformer);
```

Comment:
I deleted the last comment, but yeah I think this is the case.  recordTransformer is always null here because we don't set the member until L735.  We need to bump it up, or, just forgo the member variable and always read the config.